gammapy/time/lombscargle_gammapy.py

```diff
@@ -0,0 +1,171 @@
+
+    return t_win, window
+
+def _bootstrap(t, mag, dmag, freq, N_bootstraps):
+    rand = np.random.RandomState(42)
+    FAP = 0.05
+    N_bootstraps = 100
+    max_periods = np.empty(N_bootstraps)
+    for m in range(N_bootstraps):
+        ind = rand.randint(0, len(mag), len(mag))
+        PLS_boot = LombScargle(t, mag[ind], dmag[ind]).power(freq)
+        max_periods[m] = np.max(PLS_boot)
+
+    return max_periods
+
+def _grid(t, K):
+    max_period = (np.max(t) - np.min(t)) / 10 # see Halpern
+    min_period = 2 * np.median(t[1:-1] - t[0:-2])    
+    d_p = min_period / K    
+    n_periods = max_period / d_p
+    periods = np.linspace(min_period, max_period, n_periods)
+    grid = 1. / periods
+
+    return grid
+
+def _significance(t, mag, dmag, freq, PLS, FAP, N_bootstraps):
+    percentile = 1 - FAP
+    # compute pre-defined Beta-Distribution
+    a = (3-1)/2
+    b = (len(t)-3)/2
+    quant_pre = beta.ppf(percentile**(1./len(freq)), a, b)
+    
+    # compute CvM-minimization-fitted Beta-Distribution
+    # initial guess for CvM-minimization
+    theta_1 = -1. * (np.mean(PLS) * (-np.mean(PLS) + np.mean(PLS)**2 + np.var(PLS))) / (np.var(PLS))
+    if theta_1 < 0:
+        theta_1 = 0.00001
+    theta_2 = (theta_1 - theta_1 * np.mean(PLS)) / np.mean(PLS)
+    if theta_2 < 0:
+        theta_2 = 0.00001
+    x0 = [theta_1, theta_2]
+    cvm_minimize = optimize.fmin(_cvm, [theta_1, theta_2], args=(PLS,))
+    quant_cvm = beta.ppf(percentile**(1./len(freq)), cvm_minimize[0], cvm_minimize[1])
+    
+    # compute NLL-minimization-fitted Beta-Distribution
+    nll_minimize = optimize.fmin(_nll, [a, b], args=(PLS,))
+    quant_nll = beta.ppf(percentile**(1./len(freq)), nll_minimize[0], nll_minimize[1])
+
+    # bootstrap
+    max_periods = _bootstrap(t, mag, dmag, freq, N_bootstraps)
+    sig = np.percentile(max_periods, 100*percentile, axis=0)
+
+    return quant_pre, quant_cvm, quant_nll, sig
+
+def lomb_scargle(t, mag, dmag, K, N_bootstraps, FAP):
+    """This function processes the lomb-scargle-algorithmus in autopower mode
+    """
+    ### set up lomb-scargle-algorithm    
+    freq = _grid(t, K)
+    PLS = LombScargle(t, mag, dmag).power(freq)
+    # spectral window function
+    t_win, window = _window_function(t)
+    PLS_win = LombScargle(t_win, window, 1).power(freq)
+    
+    ### significance criteria
+    quant_pre, quant_cvm, quant_nll, quant_boot = _significance(t, mag, dmag, freq, PLS, FAP, N_bootstraps)    
+    # find significant periods
+    best_period_bar = 0
+    best_period = 0
+    for f in range(len(freq)):
+        if PLS[f] > quant_boot:#     here a significance criteria has to be chosen
+            if PLS[f] > best_period_bar:
+                best_period_bar = PLS[f]
+                best_period = 1. / freq[f]
+    if best_period == 0:
+        best_period = np.nan
+    
+    return freq, PLS, best_period, quant_pre, quant_cvm, quant_nll, quant_boot, PLS_win
```

Comment:
Here you return a tuple of 8 computed results.
People will never get that order right.

Suggest to return a `dict` instead with that info (and then document the key names and values in the docstring)